flake.nix

```diff
@@ -0,0 +1,53 @@
+{
+  description = "Flake for Plutus";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?rev=7d71001b796340b219d1bfa8552c81995017544a";
+    haskell-nix.url = "github:input-output-hk/haskell.nix?rev=87084d65a476cc826a0e8c5d281d494254f5bc7a";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, flake-utils, ... }:
+    (flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+      let
+        #
+        # Obtain all niv sources extending it with the 'nixpkgs' flake input
+        #
+        sources =
+          let
+            sourcesInfo = builtins.fromJSON (builtins.readFile ./nix/sources.json);
+            fetch = sourceInfo: builtins.fetchTarball { inherit (sourceInfo) url sha256; };
+          in
+          builtins.mapAttrs (name: info: fetch info) sourcesInfo // { inherit nixpkgs; };
```

Comment:
Why aren't we importing `sources.nix`? This seems pretty fragile - we're just pretending that all `niv` sources are the same, which isn't necessarily true.

Reply:
Can't we just use `sourcesOverride`?

Reply:
I wish, trying to import `sources.nix` directly leads to this:

```
error: attribute 'currentSystem' missing

       at /nix/store/yn1pw9mm0qljjrmpbh7wnsdp1bks82ih-source/nix/sources.nix:160:16:

          159|     , sources ? if isNull sourcesFile then {} else builtins.fromJSON (builtins.readFile sourcesFile)
          160|     , system ? builtins.currentSystem
             |                ^
          161|     , pkgs ? mkPkgs sources system
(use '--show-trace' to show detailed location information)
```

No `currentSystem` when doing flake magic.

Reply:
Looks like we can fix that by passing stuff into `sources.nix` when we call it, though.

Reply:
yes, indeed just passing `system` fixes that and we can import `sources.nix` directly without this hack.